### ISSUE TITLE
Problem with rotation when adding images

### ISSUE THREAD

**Issue by ghost** [python-openxml/python-docx#540]
I'm adding some images to a document and some of them rotate itself. The picture should be placed as portrait and it rotates automatically to landscape.

Here it is the code I'm using:

    document.add_picture(path_to_picture, width=Inches(3))
    last_paragraph = document.paragraphs[-1]
    last_paragraph.alignment = WD_ALIGN_PARAGRAPH.CENTER

It works on a for loop, so lots of images can be added.

**Comment by scanny** [comment]
What file-type images are they? JPG, or PNG perhaps? Also, how do they open up when you use a photo viewer/app on your computer (not a phone or tablet)?

**Comment by elenabdc** [comment]
I'm using .JPG files. When I open this photos in the viewer (using the default photo viewer in Windows and the default photo viewer in Debian) they appear in portrait mode, but when I open the docx document after using the code above, I see them in landscape mode.

**Comment by scanny** [comment]
If you insert them into PowerPoint by hand (not using python-pptx), do they appear with the rotation you expect?

**Comment by scanny** [comment]
The answer to the "what does PowerPoint do?" question will be interesting, but I think in any case, you're going to need to pre-process your images to "auto-rotate" them based on the EXIF-orientation value, if present. There's a stackoverflow question that seems to have a solution using PIL here:  
https://stackoverflow.com/questions/13872331/rotating-an-image-with-orientation-specified-in-exif-using-python-without-pil-in

Let us know how you go. I'm sure others will find their way here with a similar problem.

**Comment by haobaod** [comment]
yes, i've find myself in totally same situation. to the question "what does PowerPoint do?": it's portrait when i insert by hand, but when by python-pptx, it's landscape. copilot tell me it's because python-pptx and PowerPoint do not respect EXIF orientation—they insert the raw pixel data as-is.

**Comment by TheZ3ro** [comment]
apparently the answer to the question "what does PowerPoint do?" is that it interprets the Exif metadata of the image and set the `a:xfrm` `rot`, `flipV`, and `flipH` attributes.

I've opened a PR to implement this behavior here: https://github.com/python-openxml/python-docx/pull/1575

This can be reproduced with the following script:
```python
from docx import Document
from PIL import Image, ImageDraw

# Draw a black arrow shaft and arrowhead
img = Image.new("RGB", (300, 400), "white")
d = ImageDraw.Draw(img)
cx, shaft_top, shaft_h = 150, 50, 200
d.rectangle([(145, shaft_top), (155, shaft_top + shaft_h)], fill="black")
d.polygon([(150, 30), (130, 70), (170, 70)], fill="black")
# Save the image
img.save("arrow.jpg")

# Create a document and add the image
document = Document()
document.add_picture("arrow.jpg")
document.save("arrow_up.docx")

# Rotate the image only by changing the EXIF orientation, then save it
img = Image.open("arrow.jpg")
exif = img.getexif()
exif[0x0112] = 6
img.save("arrow_rotated.jpg", exif=exif)

# Create a document and add the rotated image
document = Document()
document.add_picture("arrow_rotated.jpg")
document.save("arrow_rotated.docx")
```

This is documented:
- https://learn.microsoft.com/en-us/dotnet/api/documentformat.openxml.linq.a.xfrm?view=openxml-3.0.1
- https://ooxml.org/drawingml/a-xfrm/
- https://github.com/openxml/openxml-drawingml/blob/eb31724f698312dbfa3595bc59557d1fbef41a9c/lib/openxml/drawingml/properties/transform.rb#L6-L15
